### ISSUE TITLE
Race Condition when initializing an empty Database on multiple nodes

### ISSUE THREAD

**Issue by Ionas208** [liquibase/liquibase#5254]
### Search first

- [X] I searched and no similar issues were found

### Description

When Liquibase is checking for the DATABASECHANGELOG and DATABASECHANGELOGLOCK Tables it is possible that when starting an application that uses the same Schema on multiple nodes the DATABASECHANGELOG Table creation fails because of the following sequence:
1. Both nodes check for the existence of the DATABASECHANGELOG table at roughly the same time
2. One node executes the Statement to create the table first
3. The other node fails because the table is already created

I believe this issue arises because the check for the DATABASECHANGELOG table is done before the DATABASECHANGELOGLOCK table is checked for its existence and created, and therefore there can't be any locks to prevent this race condition.

I think the bug is here: Liquibase.java:1019 checkLiquibaseTables. In this [issue](https://github.com/liquibase/liquibase/issues/2083) the creation of the DATABASECHANGELOGLOCK was made thread safe. The DATABASECHANGELOGLOCK table would need to be created before the DATABASECHANGELOG and a lock before the ChangeLogHistoryService is initialized is then necessary.

### Steps To Reproduce

This issue is hard to reproduce because it relies on a very narrow time window (nonetheless it can happen). I wasn't able to reproduce it after it failed on an internal pipeline, but I am reasonably sure that the issue is what is described above. In theory however you should be able to use the same method as described in the issue linked above, it will just happen less frequently.

### Expected/Desired Behavior

The creation of the DATABASECHANGELOG should be done under a lock.

### Liquibase Version

4.23.1

### Database Vendor & Version

Oracle 19

### Liquibase Integration

spring boot

### OS and/or Infrastructure Type/Provider

Linux

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by kevin-atx** [comment]
@filipelautert Could you review this issue and see if you agree? Here is the key point in this ticket

```
I believe this issue arises because the check for the DATABASECHANGELOG 
table is done before the DATABASECHANGELOGLOCK table is checked for its 
existence and created, and therefore there can't be any locks to prevent this
race condition.
```

**Comment by filipelautert** [comment]
@kevin-atx it makes sense - we are creating tables and validating checksums without proper database lock. I just wonder how we didn't notice any problems here before as this has been around since 2013 - maybe because the problem only shows up in a initial liquibase execution when we don't have dbcl table?

**Comment by MalloD12** [comment]
> @kevin-atx it makes sense - we are creating tables and validating checksums without proper database lock. I just wonder how we didn't notice any problems here before as this has been around since 2013 - maybe because the problem only shows up in a initial liquibase execution when we don't have dbcl table?

Might this be a good question for @nvoxland? Or should we prioritize this for further analysis? cc: @kevin-atx @filipelautert

**Comment by filipelautert** [comment]
This issue may have been fixed by https://github.com/liquibase/liquibase/pull/6218 , that will be released with 4.29.2 next week. 
If the problem remains after this release please reopen this issue.
